src/cli/llm/type_refs.rs

```diff
@@ -0,0 +1,72 @@
+use indexmap::IndexMap;
+
+use crate::core::config::Config;
+
+/// A mapping of type names to their referenced fields and the number of times
+/// each type is referenced.
+///
+/// Structure maintains a map where each key is a type name, and the
+/// corresponding value is another map that tracks the field names and the count
+/// of how many times the type name was present in the configuration.
+pub struct TypeUsageIndex<'a> {
+    type_refs: IndexMap<&'a str, IndexMap<&'a str, u32>>,
+}
+
+impl<'a> TypeUsageIndex<'a> {
+    pub fn new(config: &'a Config) -> Self {
+        let type_refs = config
+            .types
+            .keys()
+            .map(|type_name| {
+                let type_references = config
+                    .types
+                    .values()
+                    .flat_map(|t_| &t_.fields)
+                    .filter(|(_, field_)| field_.type_of.as_str() == type_name.as_str())
+                    .fold(IndexMap::new(), |mut acc, (field_name, _)| {
+                        *acc.entry(field_name.as_str()).or_insert(0) += 1;
+                        acc
+                    });
+
+                (type_name.as_str(), type_references)
+            })
+            .collect();
+
+        Self { type_refs }
+    }
+
+    pub fn get(&self, type_name: &str) -> Option<&IndexMap<&str, u32>> {
```

Comment:
```suggestion
    pub fn usage_count(&self, type_name: &str) -> &IndexMap<&str, u32> {
```

Return an empty IndexMap if it's not used anywhere.

Reply:
since we are returning an map, using `usage_map` as method name